Frameworks/AddressBook/ABAddressBook.mm

```diff
@@ -16,44 +16,87 @@
 
 #import <AddressBook/ABAddressBook.h>
 
-#import <StubReturn.h>
 #import "AssertARCEnabled.h"
+#import <StubReturn.h>
+
+#import "_ABAddressBook.h"
+
+#import "UWP/InteropBase.h"
+#import "UWP/WindowsDevicesEnumeration.h"
+#import "initguid.h"
+
+// GUID to check Windows Contacts permissions.
+DEFINE_GUID(_ABAddressBookContactsGUID, 0x7D7E8402, 0x7C54, 0x4821, 0xA3, 0x4E, 0xAE, 0xEF, 0xD6, 0x2D, 0xED, 0x93);
 
 const CFStringRef ABAddressBookErrorDomain = static_cast<const CFStringRef>(@"ABAddressBookErrorDomain");
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 ABAddressBookRef ABAddressBookCreate() {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return ABAddressBookCreateWithOptions(NULL, NULL);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 ABAddressBookRef ABAddressBookCreateWithOptions(CFDictionaryRef options, CFErrorRef* error) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    _ABAddressBook* addressBook = [[_ABAddressBook alloc] init];
+    if (addressBook.contactStore == nil) {
+        // There was an error setting up the backing
+        // contact store, so just return NULL to signify
+        // the error.
+        if (error != NULL) {
+			// In this case, the caller is responsible to release error.
+            *error = CFErrorCreate(NULL, ABAddressBookErrorDomain, kABOperationNotPermittedByUserError, NULL);
+        }
+        return NULL;
+    } else {
+        return (__bridge_retained ABAddressBookRef)addressBook;
```

Comment:
To follow the create rule ([https://developer.apple.com/library/ios/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html](https://developer.apple.com/library/ios/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html)) and give the caller ownership, since we are expecting them to call CFRelease when they are done

---
In reply to: [73208539](https://github.com/Microsoft/WinObjC/pull/786#discussion_r73208539) [](ancestors = 73208539)